### DISCUSSION TITLE
How to disable line chart toggling

### DISCUSSION THREAD

**Discussion by SkylerMime** [primefaces/community#2514] [PrimeFaces] [answered]
In my <p:lineChart> element there is only one dataset/line. It has a label that can be toggled on and off by clicking, but I want it to be always on with no label.

I tried removing the setLabel function but it just replaced it with "undefined."

Thank you for your help!

```xhtml
<p:lineChart id="netVolumeChart"
		model="#{chartViewBean.lineModel}"
		style="width: 100%; height: 500px;" />
```

```Java
public void createLineModel() {
		lineModel = new LineChartModel();
		ChartData data = new ChartData();

		LineChartDataSet dataSet = new LineChartDataSet();

		dataSet.setData(sampleData);
		dataSet.setFill(false);
		dataSet.setLabel("Sample Data");
		dataSet.setBorderColor("rgb(75, 192, 192)");
		dataSet.setTension(0.1);
		data.addChartDataSet(dataSet);

		List<String> labels = sampleLabels;
		data.setLabels(labels);

		lineModel.setData(data);
	}
```

**Comment by melloware** [comment, marked as answer]
Yep you need to use Chart Extender to turn off the legend onClick: 
https://stackoverflow.com/questions/38583894/how-to-disable-chartjs-legendclick

```js
options: {
  plugins: {
    legend: {
      onClick: null
    }  
  }
}
```


See extender docs: https://primefaces.github.io/primefaces/14_0_0/#/components/charts?id=extender

**Reply by SkylerMime** [reply]
That worked, thank you!

**Reply by melloware** [reply]
No problem!
